lib/lib.cpp

```diff
@@ -0,0 +1,246 @@
+        }
+        else if (current == "(")
+        {
+            par_counter += 1;
+            if (par_counter + counter > 1)
+                return false;
+        }
+        else if (current == ")")
+        {
+            par_counter -= 1;
+        }
+        else
+        {
+            return false;
+        }
+        if (counter < 0 || counter > 1 || par_counter < 0 || par_counter > 1)
+            return false;
+    }
+    return counter == 1 && par_counter == 0;
+}
+
+std::string infix_to_postfix_util(const std::vector<std::string>& splited_infix)
+{
+    std::stack<std::string> st;
+    std::string postfix;
+    for (const std::string& current : splited_infix)
+    {
+        if (is_operator(current))
+        {
+            while (!st.empty() && st.top() != "(" && priority(st.top()) >= priority(current))
+            {
+                postfix += st.top() + " ";
+                st.pop();
+            }
+            st.push(current);
+        }
+        else if (current == "(")
+        {
+            st.push(current);
+        }
+        else if (current == ")")
+        {
+            while (!st.empty() && st.top() != "(")
+            {
+                postfix += st.top() + " ";
+                st.pop();
+            }
+            if (st.empty())
+            {
+                return "-";
+            }
+
+            st.pop();
+        }
+        else
+        {
+            postfix += current + " ";
+        }
+    }
+    while (!st.empty())
+    {
+        postfix += st.top() + " ";
+        st.pop();
+    }
+    return postfix;
+}
+
+std::string splited_to_line(const std::vector<std::string>& splited)
+{
+    std::string line;
+    for (const std::string& current : splited)
+    {
+        line += current + " ";
+    }
+    return line;
+}
+
+std::string postfix_to_infix_util(const std::vector<std::string>& splited_postfix)
+{
+    std::stack<std::string> st;
+    std::string op;
+    int last_operation_priority = 2;
+    for (const std::string& current : splited_postfix)
+    {
+        if (is_operator(current))
+        {
+            op = st.top();
+            st.pop();
+            if (last_operation_priority < priority(current) && op.find(' ') != std::string::npos)
+            {
+                op = "(" + op + ")";
+            }
+            op = st.top() + " " + current + " " + op;
+            st.pop();
+            st.push(op);
+            last_operation_priority = priority(current);
+        }
+        else
+        {
+            st.push(current);
+        }
+    }
+    return st.top();
+}
+
+void infix_to_postfix(const std::string& infix)
+{
+    std::vector<std::string> splited_infix = split(infix);
+
+    if (!is_valid_infix(splited_infix))
+    {
+        std::cout << "Wrong input" << std::endl;
+        return;
+    }
+
+    std::string postfix = infix_to_postfix_util(splited_infix);
+
+    std::cout << postfix << std::endl;
+}
+
+void postfix_to_infix(const std::string& postfix)
```

Comment:
`postfix_to_infix` and  `postfix_to_infix_util`  seems to be the same thing. Rename

Reply:
++